src/main/kotlin/org/pkl/intellij/packages/PklProjectService.kt

```diff
@@ -307,25 +305,22 @@ class PklProjectService(private val project: Project) :
     return PklProject(metadata, projectDeps)
   }
 
-  private fun discoverProjectFiles(projectDir: VirtualFile?): List<VirtualFile> {
-    val resolvedProjectDir = projectDir ?: project.guessProjectDir() ?: return emptyList()
-    val fut = CompletableFuture<List<VirtualFile>>()
-    runBackgroundableTask("Discover PklProject files") {
-      val result = SmartList<VirtualFile>()
-      Files.walkFileTree(
-        resolvedProjectDir.toNioPath(),
-        object : SimpleFileVisitor<Path>() {
-          override fun visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult {
-            if (file.name == PKL_PROJECT_FILENAME) {
-              result.add(LocalFileSystem.getInstance().findFileByNioFile(file))
+  private fun discoverProjectFiles(): List<VirtualFile> {
+    return project.modules
+      .toList()
+      .flatMap { ModuleRootManager.getInstance(it).contentRoots.toList() }
+      .flatMap { contentRoot ->
+        // no need for async here because we are already inside a background task.
+        contentRoot.refresh(false, true)
+        buildList {
+          VfsUtil.processFileRecursivelyWithoutIgnored(contentRoot) { file ->
+            if (file.fileType == PklFileType && file.name == PKL_PROJECT_FILENAME) {
+              add(file)
             }
-            return FileVisitResult.CONTINUE
+            true
```

Comment:
Redid this: this uses the IJ APIs for walking the project tree, instead of using `java.nio`. This allows us to skip files that are ignored (e.g. by `.gitignore`), and probably also is friendlier to IJ.

Also: remove background task because this is already called from within a background task to start with.